File: app/src/main/java/com/example/muvitracker/repo/TraktApi.java
package com.example.muvitracker.repo;

// *** https://trakt.docs.apiary.io/#introduction/create-an-app

// API
// production -> https://api.trakt.tv/
// stagging -> https://api-staging.trakt.tv/ - ????
// mocky -> https://private-anon-5a4b7269e2-trakt.apiary-mock.com/ + seconda parte


// PRODUCTION
// GET https://api.trakt.tv/movies/popular
// GET https://api.trakt.tv/movies/trending - Returns all movies being watched right now. Movies with the most users are returned first.
// GET https://api.trakt.tv/movies/boxoffice - Returns the most popular movies. Popularity is calculated using the rating percentage and the number of ratings.
// BoxOffice  - Il botteghino del cinema
// GET https://api.trakt.tv/movies/id - Returns a single movie's details.
// GET https://api.trakt.tv/movies/id/aliases - nomi in altre lingue


//-> /movies/tron-legacy-2010?extended=full


import com.example.muvitracker.repo.dto.BoxofficeDto;
import com.example.muvitracker.repo.dto.DetailsDto;
import com.example.muvitracker.repo.dto.PopularDto;
import com.example.muvitracker.repo.dto.search.SearchDto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

// @GET("movies/boxoffice") - le annotazioni sono metadati


public interface TraktApi {


    //              1°STEP
    @GET("movies/popular")
    // @GET fa riferimento al metodo sotto
    Call<List<PopularDto>> getPopularMovies();


    @GET("movies/boxoffice")
        // !!! sono metadati
    Call<List<BoxofficeDto>> getBoxofficeMovies();


    //               3°STEP
    // path mobile (da mettere nei parametri) - OK
    @GET("movies/{movie_id}?extended=full")
    Call<DetailsDto> getDetailsDto(@Path("movie_id") int movieId);


    //               4°STEP
    // query mobile - OK
    @GET("search/movie,show")
    Call<List<SearchDto>> getSearch(@Query("query") String searchString);


}

File: app/src/main/java/com/example/muvitracker/repo/kotlin/TraktApi.kt
package com.example.muvitracker.repo.kotlin

import com.example.muvitracker.repo.kotlin.dto.BoxoDto
import com.example.muvitracker.repo.kotlin.dto.DetaDto
import com.example.muvitracker.repo.kotlin.dto.PopuDto
import com.example.muvitracker.repo.kotlin.dto.search.SearDto
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query


//  l'interfaccia Kotlin convertita rimane identica alla versione Java fornita,

// OK

interface TraktApi {


    // 1.
    @GET("movies/popular") // !!! @SonoMetadati
    fun getPopularMovies()
            : Call<List<PopuDto>>


    // 2.
    @GET("movies/boxoffice")
    fun getBoxofficeMovies()
            : Call<List<BoxoDto>>


    // 3. Path mobile - passa il paramentro in {}
    @GET("movies/{movie_id}?extended=full")
    fun getDetailsOfDto(@Path("movie_id") traktMovieId: Int)
            : Call<DetaDto>
    // passo parametro a -> GET {"..."}


    // 4. Query mobile
    // - dopo diventa search/movie,show?query=stringQuery
    // - GET(""+ @Query) ---> "url_base" + " "
    @GET("search/movie,show")
    fun getSearch(@Query("query") searchString: String) // query=searchString
            : Call<List<SearDto>>


}